# This notebook will show the initial data exploration of the first produced solutions
For the initial data exploration we'll look at the produced optimized solutions for our client. Based on this we will define certain thresholds that are our interests and thus hard boundaries during deciding on policy making.

In [1]:
import pandas as pd

In [4]:
#load and print the 1641 solutions found during first optimization iteration (nfe = 100.000)
results = pd.read_csv('first_optimization_results.csv')
results

Unnamed: 0  A.1 Total Costs  A.1_Expected Number of Deaths  \
0              0     3.269490e+07                       0.000000   
1              1     0.000000e+00                       0.000000   
2              2     0.000000e+00                       0.000000   
3              3     1.007149e+08                       0.000000   
4              4     0.000000e+00                       0.000000   
...          ...              ...                            ...   
1636        1636     1.568563e+08                       0.001154   
1637        1637     1.645920e+08                       0.000000   
1638        1638     8.851169e+07                       0.000000   
1639        1639     7.709999e+07                       0.000000   
1640        1640     1.927303e+08                       0.000000   

      A.2 Total Costs  A.2_Expected Number of Deaths  RfR Total Costs  \
0        1.961173e+07                       0.022584     7.028000e+08   
1        1.100325e+08                       0.004064     4.004000e+08   
2        1.961173e+07                       0.022584     6.182000e+08   
3        3.601735e+07                       0.041498     4.663000e+08   
4        1.971471e+07                       0.002738     6.199000e+08   
...               ...                            ...              ...   
1636     9.289346e+07                       0.003387     1.519000e+08   
1637     5.941010e+07                       0.000000     0.000000e+00   
1638     9.014392e+07                       0.000000     1.513800e+09   
1639     2.944400e+07                       0.034048     1.055300e+09   
1640     4.784270e+07                       0.000000     4.080000e+08   

      Expected Evacuation Costs  
0                      0.000000  
1                      0.000000  
2                      0.000000  
3                      0.000000  
4                   7143.302746  
...                         ...  
1636                3516.966218  
1637                   0.000000  
1638                   0.000000  
1639                   0.000000  
1640                   0.000000  

[1641 rows x 7 columns]

# Key interests as defined by our actor
1. Minimization of flood risk in general, but focus on farmers
1. If RfR occurs, farmer should be compensated
1. We should have lower flood risk than dike ring 4
1. Livelihoods is more important than empircal evidence


In [7]:
results.mean()

Unnamed: 0                       8.200000e+02
A.1 Total Costs                  1.032965e+08
A.1_Expected Number of Deaths    1.231961e-02
A.2 Total Costs                  8.888093e+07
A.2_Expected Number of Deaths    1.606140e-02
RfR Total Costs                  4.090143e+08
Expected Evacuation Costs        9.923107e+02
dtype: float64

# Setting boundaries and expectations
We think that livelihood and casaulties are most important so we'll take the lowest 20% of those values into consideration as candidate solutions.

In [8]:
candidates = results.nsmallest(328, ['A.2_Expected Number of Deaths', 'A.1_Expected Number of Deaths'])
candidates

Unnamed: 0  A.1 Total Costs  A.1_Expected Number of Deaths  \
12            12     0.000000e+00                            0.0   
14            14     1.476035e+08                            0.0   
18            18     1.641019e+08                            0.0   
33            33     0.000000e+00                            0.0   
35            35     4.784795e+07                            0.0   
...          ...              ...                            ...   
544          544     6.813900e+07                            0.0   
1400        1400     0.000000e+00                            0.0   
1121        1121     3.269490e+07                            0.0   
1551        1551     8.906825e+07                            0.0   
904          904     1.172995e+08                            0.0   

      A.2 Total Costs  A.2_Expected Number of Deaths  RfR Total Costs  \
12       1.967735e+08                       0.000000     9.648000e+08   
14       1.860509e+08                       0.000000     7.853000e+08   
18       1.673890e+08                       0.000000     5.540000e+08   
33       2.235129e+08                       0.000000     3.638000e+08   
35       1.994335e+08                       0.000000     3.945000e+08   
...               ...                            ...              ...   
544      2.014081e+08                       0.001474     3.390000e+08   
1400     1.836688e+08                       0.001474     5.951000e+08   
1121     4.875576e+07                       0.001478     2.792000e+08   
1551     4.878287e+07                       0.001485     1.210600e+09   
904      9.136726e+07                       0.001487     0.000000e+00   

      Expected Evacuation Costs  
12                     0.000000  
14                     0.000000  
18                   837.719329  
33                     0.000000  
35                     0.000000  
...                         ...  
544                    0.000000  
1400                   0.000000  
1121                6619.011160  
1551                1180.907186  
904                    0.000000  

[328 rows x 7 columns]

This gives us an upper boundary of:
1. `'A.1_Expected Number of Deaths'` = 0 
2. `'A.2_Expected Number of Deaths'` = 0.001487

Now that we have defined a new solution set of 328 candidate solutions based on our best 20% performing scenario's we can reevaluate the costs attatched to this.
